fuel-vm/src/interpreter/flow.rs

```diff
@@ -627,39 +616,47 @@ where
 fn write_call_to_memory<S>(
     frame: &CallFrame,
     frame_bytes: Vec<u8>,
-    code_mem_range: MemoryRange,
-    memory: &mut [u8; MEM_SIZE],
+    fp: Reg<FP>,
+    len: usize,
+    memory: &mut Memory,
     storage: &S,
 ) -> IoResult<Word, S::Error>
 where
     S: StorageSize<ContractsRawCode> + StorageRead<ContractsRawCode> + StorageAsRef,
 {
-    let mut code_frame_range = code_mem_range.clone();
     // Addition is safe because code size + padding is always less than len
-    code_frame_range.shrink_end(frame.code_size() + frame.code_size_padding());
-    code_frame_range
-        .clone()
-        .write(memory)
+    let frame_len_with_padding = frame.code_size() + frame.code_size_padding();
+    let content_size = len.saturating_sub(frame_len_with_padding);
+    memory
+        .write_noownerchecks(*fp, content_size)?
         .copy_from_slice(&frame_bytes);
 
-    let mut code_range = code_mem_range.clone();
-    code_range.grow_start(CallFrame::serialized_size());
-    code_range.shrink_end(frame.code_size_padding());
+    let code_start = fp.saturating_add(CallFrame::serialized_size() as Word);
+    let code_len = frame.code_size();
```

Comment:
The old version constructed`code_range` essentially like this

```rust
code_frame_range.shrink_end(frame.code_size() + frame.code_size_padding());
let mut code_range = code_mem_range.clone();
code_range.grow_start(CallFrame::serialized_size());
code_range.shrink_end(frame.code_size_padding());
```

Note the `shrink_end` which is used to remove the padding. So here we never add the padding, instead of subtracting it away after adding it.

Reply:
Oh, I see=D